docs/introduction/Sidechain-SDK-extension.rst

```diff
@@ -306,22 +306,22 @@ ApplicationState:
 
 The next custom state checks could be done here:
 
-  * ``public boolean validate(SidechainStateReader stateReader, SidechainBlock block)`` --  any custom block validation could be done here. If the function returns false, then the block will not be accepted by the Sidechain Node.
+  * ``public boolean validate(SidechainStateReader stateReader, SidechainBlock block)`` -- any custom block validation could be done here. If the function returns false, then the block will not be accepted by the sidechain node.
   
-  * ``public boolean validate(SidechainStateReader stateReader, BoxTransaction<Proposition, Box<Proposition>> transaction)`` -- any custom checks for the transaction could be done here if the function returns false then transaction is assumed as invalid and for example will not be included in a memory pool. 
+  * ``public boolean validate(SidechainStateReader stateReader, BoxTransaction<Proposition, Box<Proposition>> transaction)`` -- any custom checks for the transaction could be done here. If the function returns false, then the transaction is assumed as invalid and, for example, will not be included in a memory pool. 
 
   * ``public Try<ApplicationState> onApplyChanges(SidechainStateReader stateReader, byte[] version, List<Box<Proposition>> newBoxes, List<byte[]> boxIdsToRemove)`` -- any specific action after block applying in State could be defined here.
   
-  * ``public Try<ApplicationState> onRollback(byte[] version)`` -- any specific action after rollback of State (for example in case of fork/invalid block) could be defined here
+  * ``public Try<ApplicationState> onRollback(byte[] version)`` -- any specific action after rollback of State (for example, in case of fork/invalid block) could be defined here.
   
 Application Wallet 
 ##################
 
-The Wallet by default keeps user secret info and related balances. The actual data is updated when a new block is applied to the chain or when some blocks are reverted. Developers can specify custom secret types that will be processed by Wallet. The developer may extend the logic using `ApplicationWallet <https://github.com/ZencashOfficial/Sidechains-SDK/blob/master/sdk/src/main/java/com/horizen/wallet/ApplicationWallet.java>`_
+The Wallet by default keeps user secret info and related balances. The actual data is updated when a new block is applied to the chain or when some blocks are reverted. Developers can specify custom secret types that will be processed by the wallet. The developer may extend the logic using `ApplicationWallet <https://github.com/ZencashOfficial/Sidechains-SDK/blob/master/sdk/src/main/java/com/horizen/wallet/ApplicationWallet.java>`_
```

Comment:
@zennermac 

The wallet keeps user secret info and related balances b default. The data is updated when a new block is applied to the chain or when blocks are reverted. Developers can specify custom secret types that will be processed by the wallet. The developer may extend the logic using `ApplicationWallet <https://github.com/ZencashOfficial/Sidechains-SDK/blob/master/sdk/src/main/java/com/horizen/wallet/ApplicationWallet.java>`_